crates/topos-p2p/src/behaviour/gossip.rs

```diff
@@ -35,7 +37,9 @@ impl Behaviour {
     ) -> Result<usize, &'static str> {
         match topic {
             TOPOS_GOSSIP => {
-                _ = self.gossipsub.publish(IdentTopic::new(topic), message);
+                if let Ok(msg_id) = self.gossipsub.publish(IdentTopic::new(topic), message) {
```

Comment:
If we receive a `Gossip` message at minute `M` and put it in the `precedence_pool` and start delivering it at `M+2`, the `publish` method will republish it on the network. The default configuration of the key tracking is 1 min TTL. I don't think it is an issue, but we need to keep in mind that the message IDs are cached in an LRU, which doesn't last long.

Reply:
Ah, this is the `duplicate_cache_time` (https://docs.rs/libp2p/latest/libp2p/gossipsub/struct.Config.html#method.duplicate_cache_time). The size is not bounded, and the TTL is 1min by default.

Reply:
Yes, the `duplicate_cache` LRU is used in to define if a message should be sent to the network. It can be changed but the idea remains, some certificate could be gossip more than once.

Reply:
Ok I didn't have that in mind

Still better than before where we could potentially have had all peers pushing their instance of the payload more than once, while now it would only be the initial publisher attempting to publish it again, right?